rest/src/test/java/com/yahoo/ycsb/webservice/rest/IntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2016 YCSB contributors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.webservice.rest;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.ServletException;
+
+import org.apache.catalina.Context;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.startup.Tomcat;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.Assertion;
+import org.junit.contrib.java.lang.system.ExpectedSystemExit;
+import org.junit.runners.MethodSorters;
+
+import com.yahoo.ycsb.Client;
+import com.yahoo.ycsb.DBException;
+
+/**
+ * Integration test cases to verify the end to end working of the rest-binding
+ * module. It performs these steps in order. 1. Runs an embedded Tomcat
+ * server with a mock RESTFul web service. 2. Invokes the {@link Client} 
+ * class with the required parameters to start benchmarking the mock REST
+ * service. 3. Compares the response stored in the output file by {@link Client}
+ * class with the response expected. 4. Stops the embedded Tomcat server.
+ * Cases for verifying the handling of different HTTP status like 2xx & 5xx have
+ * been included in success and failure test cases.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class IntegrationTest {
+
+  @Rule
+  public final ExpectedSystemExit exit = ExpectedSystemExit.none();
+
+  private static int port = 8080;
+  private static Tomcat tomcat;
+  private static final String WORKLOAD_FILEPATH =  IntegrationTest.class.getClassLoader().getResource("workload_rest").getPath();
+  private static final String TRACE_FILEPATH = IntegrationTest.class.getClassLoader().getResource("trace.txt").getPath();
+  private static final String ERROR_TRACE_FILEPATH = IntegrationTest.class.getClassLoader().getResource("error_trace.txt").getPath();
+  private static final String RESULTS_FILEPATH = IntegrationTest.class.getClassLoader().getResource("").getPath() + "results.txt";

```

Comment:
The file results doesn't exist hence we can't call `getResource("results.txt");`. Since we need the folder path to /src/test/resources even `getResource("resources")` doesn't work. Hence I've used a period - '.' signifying the current target directory.